smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -503,6 +505,22 @@ private void writeShapeValidator(TypeScriptWriter writer,
                         });
                     }
 
+                    if (shape.isEnumShape()) {
```

Comment:
In smithy-typescript, the code generator does not convert all enum traits to enum shapes, so we have to check the EnumTrait like before.

Reply:
Fixed in [dc90a62](https://github.com/awslabs/smithy-typescript/pull/713/commits/dc90a62abc4bea71c11032154ded1196f27c5a3b).